### PR TITLE
only retry terminated builds

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/steps/source.go [hunk 1]
```diff
@@ -242,7 +242,10 @@ func handleBuild(buildClient BuildClient, build *buildapi.Build, dry bool, artif
 				return fmt.Errorf("could not get build %s: %v", build.Name, err)
 			}
 
-			if isInfraReason(b.Status.Reason) || hintsAtInfraReason(b.Status.LogSnippet) {
```
Comment (stevekuznetsov) [inline pkg/steps/source.go]: Can you make a helper `isTerminated` for readability here

File: pkg/steps/source.go [hunk 2]
```diff
@@ -301,7 +304,6 @@ func isInfraReason(reason buildapi.StatusReason) bool {
 		buildapi.StatusReasonNoBuildContainerStatus,
 		buildapi.StatusReasonFailedContainer,
 		buildapi.StatusReasonOutOfMemoryKilled,
-		buildapi.StatusReasonCannotRetrieveServiceAccount,
```
Comment (stevekuznetsov) [inline pkg/steps/source.go]: Why?

File: pkg/steps/source.go (hunk 1 above)
Comment (bparees) [reply pkg/steps/source.go]: only under protest.  one exists in origin but you don't have a dep on origin.

File: pkg/steps/source.go (hunk 2 above)
Comment (bparees) [reply pkg/steps/source.go]: it's not a fatal condition.  the controller will keep retrying the build until the serviceaccounts are created.

Comment (bparees) [reply pkg/steps/source.go]: I suppose technically the build could possibly timeout and get marked failed if the SAs never get created, in which case *maybe* this reason would still be there, i'm not sure.  if you want me to leave it, i can, but w/ the other changes in this PR you should not be seeing this exposed anymore anyway.

File: pkg/steps/source.go (hunk 1 above)
Comment (bparees) [reply pkg/steps/source.go]: updated.

Comment (stevekuznetsov) [reply pkg/steps/source.go]: I remember making an issue for you guys to expose your helpers in an importable way like a year ago ;)

Comment (bparees) [reply pkg/steps/source.go]: what's not importable about origin, hmmmmmm?

**Pull request merged**